packages/transaction-controller/src/TransactionController.ts

```diff
@@ -982,6 +982,29 @@ export class TransactionController extends BaseController<
     return { gas: estimatedGas, simulationFails };
   }
 
+  /**
+   * Estimates required gas for a given transaction and add additional gas buffer with the given multiplier.
+   *
+   * @param transaction - The transaction params to estimate gas for.
+   * @param multiplier - The multiplier to use for the gas buffer.
+   */
+  async estimateGasBuffered(
+    transaction: TransactionParams,
+    multiplier: number,
+  ) {
+    const { blockGasLimit, estimatedGas, simulationFails } = await estimateGas(
+      transaction,
+      this.ethQuery,
+    );
+
+    const gas = addGasBuffer(estimatedGas, blockGasLimit, multiplier);
+
+    return {
+      gas,
```

Comment:
@matthewwalsh0 extension gas utils return `gasLimit` here, do we need to change it as well to not breake API ?

Reply:
That was intentional to be consistent with the `estimateGas` method. We can just assign this to `gasLimit` inside the callback in the `MetaMaskController` as we do in the adoption branch.